ios/fabric/RNDateTimePickerComponentView.mm

```diff
@@ -0,0 +1,270 @@
+                    picker.preferredDatePickerStyle = UIDatePickerStyleCompact;
+                    break;
+                case RNDateTimePickerDisplayIOS::Inline:
+                    picker.preferredDatePickerStyle = UIDatePickerStyleInline;
+                    break;
+                case RNDateTimePickerDisplayIOS::Spinner:
+                    picker.preferredDatePickerStyle = UIDatePickerStyleWheels;
+                    break;
+                default:
+                    picker.preferredDatePickerStyle = UIDatePickerStyleAutomatic;
+            }
+            needsToUpdateMeasurements = true;
+        }
+    }
+    
+    if (oldPickerProps.minuteInterval != newPickerProps.minuteInterval) {
+        picker.minuteInterval = newPickerProps.minuteInterval;
+    }
+    
+    if (oldPickerProps.timeZoneOffsetInMinutes != newPickerProps.timeZoneOffsetInMinutes) {
+        // JS standard for time zones is minutes.
+        picker.timeZone = [NSTimeZone timeZoneForSecondsFromGMT:newPickerProps.timeZoneOffsetInMinutes * 60.0];
+    }
+    
+    if (oldPickerProps.accentColor != newPickerProps.accentColor) {
+        UIColor *color = RCTUIColorFromSharedColor(newPickerProps.accentColor);
+        
+        if (color != nil) {
+            [picker setTintColor:color];
+        } else {
+            if (@available(iOS 15.0, *)) {
+                [picker setTintColor:[UIColor tintColor]];
+            } else {
+                [picker setTintColor:[UIColor systemBlueColor]];
+            }
+        }
+    }
+    
+    if (oldPickerProps.textColor != newPickerProps.textColor) {
+        [self updateTextColorForPicker:picker color:RCTUIColorFromSharedColor(newPickerProps.textColor)];
+    }
+    
+    if (@available(iOS 13.0, *)) {
+        if (oldPickerProps.themeVariant != newPickerProps.themeVariant) {
+            switch (newPickerProps.themeVariant) {
+                case RNDateTimePickerThemeVariant::Light:
+                    picker.overrideUserInterfaceStyle = UIUserInterfaceStyleLight;
+                    break;
+                case RNDateTimePickerThemeVariant::Dark:
+                    picker.overrideUserInterfaceStyle = UIUserInterfaceStyleDark;
+                    break;
+                default:
+                    picker.overrideUserInterfaceStyle = UIUserInterfaceStyleUnspecified;
+            }
+        }
+    }
+    
+    if (oldPickerProps.enabled != newPickerProps.enabled) {
+        picker.enabled = newPickerProps.enabled;
+    }
+    
+    return needsToUpdateMeasurements;
+}
+
+- (void)updateProps:(Props::Shared const &)props oldProps:(Props::Shared const &)oldProps
+{
+    // Updating the dummy first to check if we need to update measurements
+    Boolean needsToUpdateMeasurements = [self updatePropsForPicker:_dummyPicker props:props oldProps:oldProps];
+    
+    if (needsToUpdateMeasurements) {
+        [self updateMeasurements];
+    }
+    
+    [self updatePropsForPicker:_picker props:props oldProps:oldProps];
```

Comment:
of course, [line 253](https://github.com/react-native-datetimepicker/datetimepicker/pull/657/files/963ea83d3017547a989e63f1102287a5dc8e53b3#diff-fc0cc2424328185b2435e4942709873bbc6a7903252d0c100a17be58908d41e1R253) updates `_dummyPicker` first.

It needs the `_dummyPicker` because I found resizing before updating has a better experience. Otherwise, if the new date is larger than the old date, it will show the compact format `mm/dd/yy` for half a second and change to `Sep 23, 2022` after the resize.

Reply:
We have the same mirror picker in the Paper implementation. In `RNDateTimePickerManager`, the [view](https://github.com/react-native-datetimepicker/datetimepicker/blob/master/ios/RNDateTimePickerManager.m#L74) stores the main picker and [_picker](https://github.com/react-native-datetimepicker/datetimepicker/blob/master/ios/RNDateTimePickerManager.m#L65) stores the "mirror" or "dummy" picker that the shadow view is going to use to calculate the size.

Reply:
Here's how it looks without the dummy picker.
https://user-images.githubusercontent.com/3791120/192032708-17dee70d-e140-4349-ac3e-4e956ea0e83b.mp4

Reply:
> [line 253](https://github.com/react-native-datetimepicker/datetimepicker/pull/657/files/963ea83d3017547a989e63f1102287a5dc8e53b3#diff-fc0cc2424328185b2435e4942709873bbc6a7903252d0c100a17be58908d41e1R253) updates _dummyPicker first.

ah, I misread that, sorry. Thanks for the gif, it's helpful!